edx_proctoring/tests/test_views.py

```diff
@@ -1213,21 +1214,25 @@ def test_reset_attempt_action(self):
         self.assertGreater(response_data['exam_attempt_id'], 0)
         old_attempt_id = response_data['exam_attempt_id']
 
-        # complete exam, then attempt to reset progress
-        for action in ['submit', 'reset_attempt']:
-            response = self.client.put(
-                reverse('edx_proctoring:proctored_exam.attempt', args=[old_attempt_id]),
-                json.dumps({
-                    'action': action,
-                }),
-                content_type='application/json'
-            )
+        # reject exam, then attempt to reset progress
+        set_runtime_service('grades', MockGradesService(rejected_exam_overrides_grade=False))
+        update_attempt_status(
+            proctored_exam.id, self.student_taking_exam.id, ProctoredExamStudentAttemptStatus.rejected
+        )
+        response = self.client.put(
```

Comment:
Does this mean it's frontend's job to reset the existing attempt into `reset_attempt` action?

Reply:
yup